### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by jing-xiang**
1. Enabled assertions for build gradle
2. Added PIN feature 
3. Added password hashing
4. Added reset password feature

### CODE REVIEW

**Comment by jing-xiang** [comment]
Enable assertions included

File: src/main/java/longah/LongAh.java [hunk 1]
```diff
@@ -46,6 +49,13 @@ public static void main(String[] args) {
         LongAhLogger.log(Level.INFO, "Starting Pre-program preparations.");
         System.out.println("Welcome to LongAh!");
         LongAh app = new LongAh();
+        PINHandler pinHandler = new PINHandler();
+
+        if (!Files.exists(Paths.get(PINHandler.getPinFilePath()))|| pinHandler.loadPin().isEmpty()) {
```
Comment (1simjustin) [inline src/main/java/longah/LongAh.java:54]: possible to have this done as part of pinHandler constructor? I'm trying to reduce clutter in main code at this point

File: src/main/java/longah/commands/ResetCommand.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    public void execute(Group group) throws LongAhException {
+        execute();
+    }
+
+    /**
+     * Executes the reset command.
+     *
+     * @throws LongAhException If unexpected additional parameters are found.
+     */
+    public void execute() throws LongAhException {
```
Comment (1simjustin) [inline src/main/java/longah/commands/ResetCommand.java:31]: uhh can u comment here on when this is called? `main` calls functions using a general call of `c.execute(group)` so not too clear on this

File: src/main/java/longah/handler/PINHandler.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    /**
+     * Creates a new PIN for the user.
+     */
+    public void createPin() {
+        System.out.println("Thanks for choosing LongAh! Never worry about owing money during the Year of the Dragon!\n"
+                + "Create your 6-digit PIN:\n");
+        String pin = scanner.nextLine();
+
+        while (pin.length() != 6 || !pin.matches("\\d{6}")) {
+            if (Objects.equals(pin, "quit")) {
```
Comment (1simjustin) [inline src/main/java/longah/handler/PINHandler.java:48]: believe our stop command is `exit` instead

File: src/main/java/longah/handler/PINHandler.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+    /**
+     * Creates a new PIN for the user.
+     */
+    public void createPin() {
+        System.out.println("Thanks for choosing LongAh! Never worry about owing money during the Year of the Dragon!\n"
+                + "Create your 6-digit PIN:\n");
+        String pin = scanner.nextLine();
+
+        while (pin.length() != 6 || !pin.matches("\\d{6}")) {
```
Comment (1simjustin) [inline src/main/java/longah/handler/PINHandler.java:47]: just for general info, can u leave a comment on what the regex signifies

File: src/main/java/longah/handler/PINHandler.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    public String loadPin() {
+        String savedPin = "";
+        try {
+            // Load hashed PIN after loading
+            savedPin = new String(Files.readAllBytes(Paths.get(PIN_FILE_PATH)));
+            logger.log(Level.INFO, "User loaded successfully!");
+        } catch (IOException e) {
+            System.out.println("Error reading saved PIN. Please try again.");
+        }
+        return savedPin;
```
Comment (1simjustin) [inline src/main/java/longah/handler/PINHandler.java:89]: err might want to save it to a variable within here and have a sep `getPin()` method so you don't have to keep accessing the file.

File: src/main/java/longah/handler/PINHandler.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        assert enteredPin != null : "Entered PIN should not be null.";
+
+        try {
+            // Hash the entered PIN before comparing
+            MessageDigest md = MessageDigest.getInstance("SHA-256");
+            byte[] hashedEnteredPin = md.digest(enteredPin.getBytes(StandardCharsets.UTF_8));
+            String hashedEnteredPinHex = new BigInteger(1, hashedEnteredPin).toString(16);
+
+            while (!hashedEnteredPinHex.equals(savedPin)) {
+                if (Objects.equals(enteredPin, "quit")) {
```
Comment (1simjustin) [inline src/main/java/longah/handler/PINHandler.java:110]: as above

File: text-ui-test/EXPECTED.TXT
Comment (1simjustin) [inline text-ui-test/EXPECTED.TXT:1]: lmao. kiv to fix

**Review by 1simjustin — APPROVED** [review]
Uhh understand god a few shifts in some of the logic. ill approve first but get it up fast fast pls n ty.
Try to use `static` type for this also, pin should be singleton.

File: src/main/java/longah/commands/ResetCommand.java (hunk 2 above)
Comment (jing-xiang) [reply src/main/java/longah/commands/ResetCommand.java:31]: called by using the command "reset password". "password" is passed in as an expression for checking

File: src/main/java/longah/handler/PINHandler.java (hunk 3 above)
Comment (jing-xiang) [reply src/main/java/longah/handler/PINHandler.java:48]: resolved

**Pull request merged**